metabase/metabase
# Update Admin User Form to use Formik

Closes https://github.com/metabase/metabase/issues/35605

### Description
This PR migrates removes the User entity forms, instead driving new user and edit user forms from a shared component that accesses Formik directly. A small summary of changes below
- new `<UserForm />` component that contains the new form, and is used by the new and edit user containers
- `<FormGroupWidget />` was moved to `metabase/forms`, converted to TS and updated to fetch groups via hook rather than entity HoC
- `<LoginAttributes>` and `<MappingEditor>` were updated to TS and now use `metabase/ui` components
- `PLUGIN_AD MIN_USER_FORM_FIELDS` is now a map of (one) components, instead of an array to push data onto
- New hooks were added for fetching group lists and users

These changes should not really impact the user experience. Forms should continue to work as they have before.

### How to verify
Go to admin -> people, edit some existing users and invite new ones (being sure to also try adjusting groups and login attributes). Ensure that login attributes are not visible in OSS

### Checklist
